tests/erc20/test_erc20_mintable_burnable.py

```diff
@@ -0,0 +1,301 @@
+    # https://docs.google.com/document/d/1YLPtQxZu1UAvO9cZ1O2RPXBbT0mooh4DYKjA_jp-RLM/edit#heading=h.m9fhqynw2xvt
+    # the final spec insists on NOT using this behavior
+    assert c.allowance(a2, a1) == 0
+    c.approve(a1, 1, transact={'from': a2})
+    assert c.allowance(a2, a1) == 1
+    c.approve(a1, 2, transact={'from': a2})
+    assert c.allowance(a2, a1) == 2
+    # Check that approving 0 then amount also works
+    c.approve(a1, 0, transact={'from': a2})
+    assert c.allowance(a2, a1) == 0
+    c.approve(a1, 5, transact={'from': a2})
+    assert c.allowance(a2, a1) == 5
+    # Check that burnFrom to ZERO_ADDRESS failed
+    assert_tx_failed(lambda: c.burnFrom(
+        ZERO_ADDRESS, 0, transact={'from': a1}))
+
+
+def test_raw_logs(c, w3, get_log_args):
+    minter, a1, a2, a3 = w3.eth.accounts[0:4]
+
+    # Check that mint appropriately emits Transfer event
+    args = get_log_args(
+        c.mint(a1, 2, transact={'from': minter}), c, 'Transfer')
+    assert args._from == ZERO_ADDRESS
+    assert args._to == a1
+    assert args._value == 2
+
+    args = get_log_args(
+        c.mint(a1, 0, transact={'from': minter}), c, 'Transfer')
+    assert args._from == ZERO_ADDRESS
+    assert args._to == a1
+    assert args._value == 0
+
+    # Check that burn appropriately emits Transfer event
+    args = get_log_args(c.burn(1, transact={'from': a1}), c, 'Transfer')
+    assert args._from == a1
+    assert args._to == ZERO_ADDRESS
+    assert args._value == 1
+
+    args = get_log_args(c.burn(0, transact={'from': a1}), c, 'Transfer')
+    assert args._from == a1
+    assert args._to == ZERO_ADDRESS
+    assert args._value == 0
+
+    # Check that transfer appropriately emits Transfer event
+    args = get_log_args(c.transfer(
+        a2, 1, transact={'from': a1}), c, 'Transfer')
+    assert args._from == a1
+    assert args._to == a2
+    assert args._value == 1
+
+    args = get_log_args(c.transfer(
+        a2, 0, transact={'from': a1}), c, 'Transfer')
+    assert args._from == a1
+    assert args._to == a2
+    assert args._value == 0
+
+    # Check that approving amount emits events
+    args = get_log_args(c.approve(a1, 1, transact={'from': a2}), c, 'Approval')
+    assert args._owner == a2
+    assert args._spender == a1
+    assert args._value == 1
+
+    args = get_log_args(c.approve(a2, 0, transact={'from': a3}), c, 'Approval')
+    assert args._owner == a3
+    assert args._spender == a2
+    assert args._value == 0
+
+    # Check that transferFrom appropriately emits Transfer event
+    args = get_log_args(c.transferFrom(
+        a2, a3, 1, transact={'from': a1}), c, 'Transfer')
+    assert args._from == a2
+    assert args._to == a3
+    assert args._value == 1
+
+    args = get_log_args(c.transferFrom(
+        a2, a3, 0, transact={'from': a1}), c, 'Transfer')
+    assert args._from == a2
+    assert args._to == a3
+    assert args._value == 0
+
+
+def test_failed_send_in_withdraw(c, w3):
```

Comment:
I think we don't need this test because it tests `withdraw` function on fallback in sending ether.

Reply:
:agree:
(Why did I add that? 🤔 )

Reply:
You are deceived by me.

Reply:
😂

Reply:
https://github.com/LayerXcom/verified-vyper-contracts/pull/53